src/compute/cast/mod.rs

```diff
@@ -324,14 +327,35 @@ fn cast_large_to_list(array: &ListArray<i64>, to_type: &DataType) -> ListArray<i
 /// * Utf8 to boolean
 /// * Interval and duration
 pub fn cast(array: &dyn Array, to_type: &DataType) -> Result<Box<dyn Array>> {
-    cast_with_options(array, to_type, CastOptions { wrapped: false })
+    cast_with_options(array, to_type, CastOptions::default())
 }
 
 /// Similar to [`cast`], but overflowing cast is wrapped
 /// Behavior:
 /// * PrimitiveArray to PrimitiveArray: overflowing cast will be wrapped (i.e. `256i16 as u8 = 0` vectorized).
 pub fn wrapping_cast(array: &dyn Array, to_type: &DataType) -> Result<Box<dyn Array>> {
-    cast_with_options(array, to_type, CastOptions { wrapped: true })
+    cast_with_options(
+        array,
+        to_type,
+        CastOptions {
+            wrapped: true,
+            partial: false,
+        },
+    )
+}
+
+/// Similar to [`cast`], but parse the utf8/binary into integer at the best-effort.
+/// Behavior:
+/// * PrimitiveArray to PrimitiveArray: overflowing cast will be wrapped (i.e. `256i16 as u8 = 0` vectorized).
```

Comment:
this is not true, right? I think we need to pass the `CastOptions` here, since we how have 4 combinations for each value of `wrapped` and `partial`. I would go ahead and change/add `pub fn cast(array, to_type, options)` and remove the other functions. This is a sufficiently important use-case to break the API. :)